### DISCUSSION TITLE
Entity State

### DISCUSSION THREAD

**Discussion by jofotoo** [jcwillox/hass-auto-backup#31] [Q&A] [answered]
Hi, i hope someone sees this :) I didn't want to make an 'issue'.

What defines the entity's state ("Pending backups")? Since the integration doesn't actively do anything except create services to call, how does this get populated?

**Comment by jcwillox** [comment, marked as answer]
Great, this is the perfect place for questions like that.

While Auto Backup doesn't create the backups itself, it still has to (and can) wait for a backup to complete. When you call an `auto_backup.snapshot_*` service, auto-backup tells the supervisor to create a snapshot and sets pending backups to 1 when the snapshot is complete auto-backup sets it back to 0.

If you try to create two snapshots at the same time you will briefly see pending backups go to 2 then the second will immediately fail and return pending backups to 1, as you can't take two snapshots at the same time.

The original idea was that at some point I'd add the ability to effectively queue snapshot requests, but I decided against it as it would likely be error-prone and add unnecessary complexity, among a few other issues with it. Really `1` should be auto-backup is creating a backup, and `0` auto-backup is not doing anything.
